assets/js/base/context/hooks/use-customer-data.ts

```diff
@@ -95,6 +95,39 @@ export const useCustomerData = (): {
 	// Store values last sent to the server in a ref to avoid requests unless important fields are changed.
 	const previousCustomerData = useRef< CustomerData >( customerData );
 
+	// Need to sync shipping and billing address from wp/store/cart to local state, because `customerData` is
+	// populated before the wc/store/cart is hydrated. We only need to run this the first time they're out of sync
+	// because subsequent times will be the result of this effect or client side changes.
+	const [ hasCustomerDataSynced, setHasCustomerDataSynced ] = useState<
+		boolean
+	>( false );
+
+	useEffect( () => {
+		if (
+			! hasCustomerDataSynced &&
```

Comment:
This might be more readble if we returned early after checking just `hasCustomerDataSynced` before the rest of the conditional.